File: GENETIKalk_FINAL/model/monomero.cpp
#include "monomero.h"

//Costruttore
Monomero::Monomero(QString n, QString s, double p) : _name(std::move(n)), _acr(s), _mass(p) {}

//Metodi di get
QString Monomero::getName() const {
    return _name;
}

QString Monomero::getAcr() const {
    return _acr;
}

double Monomero::getMass() const {
    return _mass;
}

//Metodi di set
void Monomero::setName(const QString &name) {
    _name = name;
}

void Monomero::setAcr(const QString &acr) {
    _acr = acr;
}

void Monomero::setMass(double mass) {
    _mass = mass;
}

Monomero *Monomero::clone() const {
    return new Monomero(*this);
}

bool Monomero::operator==(const Monomero &m) const {
    return this->getAcr() == m.getAcr() && this->getName() == m.getName();
}

File: JavaGenetiKalk/src/Monomero/Monomero.java

package Monomero;

public class Monomero{
    private String _name;
    private String _acr;
    private double _mass;

    public Monomero(String name, String acr, double mass){
        _name=name;
        _acr=acr;
        _mass=mass;
    }

    public String getName(){
        return _name;
    }

    public void setName(String name) {
        _name=name;
    }

    public String getAcr(){
        return _acr;
    }

    public void setAcr(String acr) {
        _acr=acr;
    }

    public double getMass(){
        return _mass;
    }

    public void setMass(double mass){
        _mass=mass;
    }
}
